honeybee_radiance/sensor.py

```diff
@@ -49,14 +51,76 @@ def from_raw_values(cls, x=0, y=0, z=0, dx=0, dy=0, dz=1):
 
     @property
     def pos(self):
-        """Get the position of the sensor as a tuple of 3 (x, y, z) numbers."""
+        """Get or set the position of the sensor as a tuple of 3 (x, y, z) numbers."""
         return self._pos
 
+    @pos.setter
+    def pos(self, value):
+        self._pos = typing.tuple_with_length(value) if value is not None else (0, 0, 0)
+
     @property
     def dir(self):
-        """Get the dir of the sensor as a tuple of 3 (x, y, z) numbers."""
+        """Get or set the dir of the sensor as a tuple of 3 (x, y, z) numbers."""
         return self._dir
 
+    @dir.setter
+    def dir(self, value):
+        self._dir = typing.tuple_with_length(value) if value is not None else (0, 0, 1)
+
+    def move(self, moving_vec):
+        """Move this sensor along a vector.
+
+        Args:
+            moving_vec: A ladybug_geometry Vector3D with the direction and distance
+                to move the sensor.
+        """
+        self.pos = tuple(pv.Point3D(*self.pos).move(moving_vec))
```

Comment:
What if we use `Point3D` and `Vector3D` instead of tuple now that you need it for all the transforms?

We can add a classmethod for creating sensors from tuples or list.

Reply:
Because you built in all of these radiance serialization capabilities for the class that rely on everything being a tuple.  If you want to re-do all of these radiance serialization functions to use the ladybug_geometry objects instead of tuples, maybe then it will make sense.  But Radiance serialization is a more common use case than transforms so, at the moment, it wasn't worth all of this effort.

Reply:
Not to mention the fact that so many of the tests would need to be changed.  It just seems like a task where, if I were to do it, there's a very high probability of breaking things.

Reply:
OK. then let's leave it as is.

Reply:
On second though, maybe I could do it easily for the SensorGrid if you can take care of it for the View class.  I just don't know how we should handle all of these properties that are `TupleOption`.